### PR TITLE
ref(cds): replace local function to reuse helper

### PR DESCRIPTION
**Pull request by SanyaKochhar**
Updates the GetMatchingServices return type to be reusable

Signed-off-by: Sanya Kochhar <kochhars@microsoft.com>

### CODE REVIEW

File: pkg/envoy/cds.go [hunk 1]
```diff
@@ -81,32 +80,6 @@ func (c HasClusterCheck) Run() error {
...
-	svcList, err := c.k8s.CoreV1().Services(c.dstPod.Namespace).List(context.Background(), metav1.ListOptions{})
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to list services in namespace %s", c.dstPod.Namespace)
-	}
-
-	for _, svc := range svcList.Items {
-		svc := svc
-		if svc.Namespace != c.dstPod.Namespace {
-			continue
-		}
```
Comment (SanyaKochhar) [inline pkg/envoy/cds.go:95]: @nojnhuh wanted to make sure it's okay that is not part of the helper being used now... the Services.List is called only on the dstPod's Namespace so this check feels a bit redundant...?

File: pkg/envoy/cds.go [hunk 2]
```diff
@@ -1,14 +1,13 @@
 package envoy
 
 import (
-	"context"
 	"fmt"
 
+	"github.com/openservicemesh/osm-health/pkg/kuberneteshelper"
```
Comment (draychev) [inline pkg/envoy/cds.go:6]: @SanyaKochhar The `/openservicemesh/` packages should be grouped in the last section of imports

**Review by draychev — APPROVED** [review]
lgtm

File: pkg/envoy/cds.go (hunk 1 above)
Comment (nojnhuh) [reply pkg/envoy/cds.go:95]: Sorry this is so late, but yes, you're right.

**Pull request merged**